rc ruby: Run ruby-trim-indent on mode change

kayhide opened the pull request:
As seen in other languages, run `*-trim-indent` when getting out of insert mode.

mawww commented:
Hello

I am very late in reviewing this, sorry about that, it seems to break the `do-keyword` test in `indent/ruby` can you check that and either update the test or the indent logic ?

kayhide commented:
Sure, I will take a look.

kayhide commented:
@mawww Hi, I have updated the test, pushed it.
The test itself looked working, but some of CI tests failed.
https://cirrus-ci.com/build/5797598868013056

Then I rebased this branch onto the current master, CI still fails.
https://cirrus-ci.com/build/5841857264484352

At least, linux tests are passing.
Does this look good?

krobelus commented:
That's a bug in master, you can cherry-pick 272b702 from #4156 to make the CI pass

kayhide commented:
@krobelus Okay, I will try it, thanks.

kayhide commented:
@mawww Hi, it seems this PR is ready to get merged.

Or am I still expected to do something?
